src/rules.rs

```diff
@@ -107,10 +107,10 @@ pub(crate) fn spacing() -> SpacingDsl {
         .inside(NODE_LET_IN).before(T![in]).when(header_has_multi_value).when(before_token_not_newline).newline()
         .inside(NODE_LET_IN).after(T![in]).when(in_body_newline).newline()
         .inside(NODE_LET_IN).before(NODE_KEY_VALUE).when(header_has_multi_value).when(before_token_not_newline).newline()
-        
+
         .test("{a?3}: a", "{ a ? 3 }: a")
         .inside(NODE_PAT_ENTRY).around(T![?]).single_space()
-        
```

Comment:
@Rizary you might want to tweak your editor to strip whitespace on save. I think in Emacs this is handled by whitespace cleanup mode: https://github.com/purcell/whitespace-cleanup-mode

Reply:
@matklad I think I have it enable. Let me check with my doom emacs configuration.